### DISCUSSION TITLE
How to disable `eval`?

### DISCUSSION THREAD

**Discussion by Rudxain** [uBlockOrigin/uBlock-issues#3370] [Q&A]
Where in the docs can I find info about this?

I want to write a scriptlet/user-script that:
- `delete`s eval, or assigns `undefined` to its binding.
- makes `setTimeout` & `setInterval` `throw` when passing anything but a `function` as call-back.

As an unsafe alternative, I would also like to know how to `eval = _ => {}`, as a last resort.

I know there's [an extension designed for this](https://mybrowseraddon.com/noeval.html), but I want to be able to white-list domains.

If there's no other choice, then I'll have to install TamperMonkey

**Comment by Yuki2718** [comment]
Is this even an uBO matter? It's not Ask me anything here, reading and answering questions takes time of volunteers.

**Reply by Rudxain** [reply]
I'm sorry if I wasted anyone's time. I noticed that static-filters could inject arbitrary (or "quasi-arbitrary") scriptlets into pages, so I was wondering if I could write the same in "My Filters"

**Comment by garry-ut99** [comment]
It's in an obvious place, on scriptlets page: https://github.com/uBlockOrigin/uBlock-issues/wiki/Resources-Library#noevaljs-
By the way, there is also: https://github.com/uBlockOrigin/uBlock-issues/wiki/Resources-Library#evaldata-prunejs-
